### PR TITLE
feat(challenge-parser): add feedback to mc questions

### PR DESCRIPTION
**Pull request by moT01**
This adds the ability to have a `feedback` heading in the multiple choice challenge markdown. e.g.

```
## --answers--

College students.

And others.

### --feedback--

No.

Well, maybe.

---

People who want to become software developers.

---

Everyone.
```

The feedback will show up when campers guess that answer. I think this will come in handy across the board. I intend to use it for the English challenges, as well as possibly the C# challenges.

I added one challenge with feedback in each i18n cause the build was failing without it.

Checklist:

- [x] I have read and followed the [contribution guidelines](https://contribute.freecodecamp.org).
- [x] I have read and followed the [how to open a pull request guide](https://contribute.freecodecamp.org/#/how-to-open-a-pull-request).
- [x] My pull request targets the `main` branch of freeCodeCamp.
- [x] I have tested these changes either locally on my machine, or GitPod.

Closes #XXXXX

### CODE REVIEW

**Review by naomi-lgbt — CHANGES_REQUESTED** [review]
I love the idea. I think, though, if that text is updating with the feedback, we should make screen readers announce it?

**Comment by moT01** [comment]
Thanks for taking a look @naomi-lgbt. Mind if we save that for another PR?

File: client/src/redux/prop-types.ts [hunk 1]
```diff
@@ -47,9 +47,14 @@ export type MarkdownRemark = {
   };
 };
 
+export type MultipleChoiceAnswer = {
+  answer: string;
+  feedback: string;
```
Comment (ShaunSHamilton) [inline client/src/redux/prop-types.ts:52]: Is there a particular reason to not do this?

```suggestion
  feedback: string | null;
```

File: tools/challenge-parser/parser/plugins/utils/before-heading.test.js [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+    expect.assertions(2);
+    const actual = getAllBefore(simpleAst, '--not-a-marker--');
+    expect(isArray(actual)).toBe(true);
+    expect(actual.length).toBe(0);
+  });
+
+  it('should include the whole AST before the marker', () => {
+    expect.assertions(1);
+    const actual = getAllBefore(simpleAst, '--hints--');
+    expect(actual.length === 6).toBe(true);
```
Comment (ShaunSHamilton) [inline tools/challenge-parser/parser/plugins/utils/before-heading.test.js:22]: Nit for better assertion when failing

```suggestion
    expect(actual).toHaveLength(6);
```

**Review by ShaunSHamilton — CHANGES_REQUESTED** [review]

File: tools/challenge-parser/parser/plugins/utils/before-heading.test.js (hunk 2 above)
Comment (moT01) [reply tools/challenge-parser/parser/plugins/utils/before-heading.test.js]: ✔️

File: client/src/redux/prop-types.ts (hunk 1 above)
Comment (moT01) [reply client/src/redux/prop-types.ts]: ✔️

**Review by naomi-lgbt — APPROVED** [review]

**Review by Sembauke — APPROVED** [review]

**Comment by Sembauke** [comment]
can't merge because translation workflow is failing

**Comment by moT01** [comment]
Are you sure it's not just cause changes are requested @Sembauke?

**Comment by Sembauke** [comment]
Yes, that's it actually. @ShaunSHamilton Could you please put in another review and merge it?

**Review by ShaunSHamilton — APPROVED** [review]

**Pull request merged**